mono/metadata/object.c

```diff
@@ -3966,25 +3967,25 @@ mono_runtime_get_main_args (void)
  * \returns a \c MonoArray with the arguments passed to the main
  * program. On failure returns NULL and sets \p error.
  */
-MonoArray*
-mono_runtime_get_main_args_checked (MonoError *error)
+MonoArrayHandle*
+mono_runtime_get_main_args_checked_handle (MonoError *error)
 {
-	MonoArray *res;
+	HANDLE_FUNCTION_ENTER ();
+	MonoArrayHandle *res;
 	int i;
 	MonoDomain *domain = mono_domain_get ();
-
 	error_init (error);
 
-	res = (MonoArray*)mono_array_new_checked (domain, mono_defaults.string_class, num_main_args, error);
-	return_val_if_nok (error, NULL);
+	res = (MonoArrayHandle*) mono_array_new_handle(domain, mono_defaults.string_class, num_main_args, error);
```

Comment:
Declarete `MonoArrayHandle res` and then just do `res = mono_array_new_handle (...)`